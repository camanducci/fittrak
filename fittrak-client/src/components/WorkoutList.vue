<template>
  <div>
    <h2>{{ title }}</h2>
    <p v-if="workouts.length">
      Workouts
      <ul>
        <WorkoutItem
         v-for="workout in workouts" 
          :key="workout.id"
          :workout="workout"
          />
      </ul>
    </p>
    <p v-else>
      No workouts!
    </p>
  </div>
</template>

<script>
import gql from "graphql-tag";
import WorkoutItem from "./WorkoutItem";

export default {
  name: "WorkoutList",

  data() {
    return {
      workouts: []
    };
  },

  apollo: {
    workouts: {
      query() {
        return gql`
          {
            viewer {
              workouts {
                id
                dateStarted
                dateEnded
                isActive
                status
                slug
                exercises {
                  id
                }
              }
            }
          }
        `;
      },

      update: data => data.viewer.workouts
    }
  },

  components: {
    WorkoutItem
  },

  props: {
    title: String
  }
};
</script>
