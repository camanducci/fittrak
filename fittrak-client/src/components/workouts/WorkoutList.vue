<template>
  <v-flex 
    mt-5 
    v-if="$apollo.loading" 
    text-xs-center>
    <v-progress-circular 
      color="primary" 
      :indeterminate="true" 
      size="48"/>
  </v-flex>
  <v-flex v-else>
    <v-flex v-if="workouts.length">
      <v-list 
        two-line 
        subheader 
        class="mt-2">
        <v-subheader v-if="title">
          {{ title }}
        </v-subheader>
        <WorkoutItem
          v-for="workout in workouts"
          :key="workout.id"
          :workout="workout"
        />
      </v-list>
    </v-flex>
    <v-flex v-else>
      <v-flex text-xs-center>No workouts available! ðŸ˜ž</v-flex>
    </v-flex>
  </v-flex>
</template>

<script>
import { getStatusBySlug } from "@/helpers";

import WorkoutItem from "@/components/workouts/WorkoutItem";

import WORKOUTS from "@/graphql/queries/workouts.graphql";

export default {
  name: "WorkoutList",

  data() {
    return {
      workouts: [],
      selectedStatus: null
    };
  },

  created() {
    // Set the status if we're currently in view
    const { status } = this.$route.params;

    if (status) {
      this.selectedStatus = getStatusBySlug(status);
    }
  },

  watch: {
    $route(to) {
      const { status } = to.params;

      if (status) {
        this.selectedStatus = getStatusBySlug(status);
      }
    }
  },

  apollo: {
    workouts: {
      query: WORKOUTS,
      variables() {
        const { status } = this.$props;

        const vars = {
          status
        };

        if (this.selectedStatus) {
          vars.status = this.selectedStatus;
        }

        return vars;
      },
      update(data) {
        const { limit } = this.$props;

        if (limit) return data.workouts.slice(0, limit);

        return data.workouts;
      }
    }
  },

  components: {
    WorkoutItem
  },

  props: {
    status: {
      type: String,
      required: false,
      default: ""
    },

    title: {
      type: String,
      required: true
    },

    limit: {
      type: Number,
      required: true
    }
  }
};
</script>

<style scoped>
</style>
