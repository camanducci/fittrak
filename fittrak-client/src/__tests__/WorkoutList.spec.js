import { shallowMount, createLocalVue } from "@vue/test-utils";
import VueRouter from "vue-router";

import WorkoutList from "@/components/workouts/WorkoutList";

describe("WorkoutList.vue", () => {
  const localVue = createLocalVue();
  localVue.use(VueRouter);
  const router = new VueRouter();

  test("WorkoutList renders", () => {
    const component = shallowMount(WorkoutList, {
      localVue,
      router,
      mocks: {
        $apollo: {
          loading: false
        }
      },
      propsData: {
        status: "PENDING",
        title: "Recently",
        limit: 5
      }
    });
    expect(component.html()).toMatchSnapshot();
  });
});
