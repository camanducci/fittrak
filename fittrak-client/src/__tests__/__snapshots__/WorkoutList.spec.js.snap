// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkoutList.vue WorkoutList renders 1`] = `
<v-flex-stub tag="div">
  <v-flex-stub tag="div">
    <v-flex-stub tag="div" text-xs-center="">No workouts available! ðŸ˜ž</v-flex-stub>
  </v-flex-stub>
</v-flex-stub>
`;
